sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -725,10 +725,19 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
    * @return
    */
   private def convertDecimal64ToDecimal32Wrapper(cv: ColumnVector, precision: Int): ColumnVector = {
+    /**
+     * This method returns a ColumnView that should be copied out to a ColumnVector before closing
+     * the `toClose` views otherwise it will close the returned view and it's children as well.
+     *
+     * @param cv          ColumnView to be checked if it has Decimal32 values stored as Decimal64
+     * @param precision   precision list of all the Decimal columns in this ColumnVector
+     * @param toClose     The ColumnViews to be closed after this method is done. This will also
+     *                    contain the returned view
+     */
     def convertDecimal64ToDecimal32(
-        cv: ColumnView,
-        precision: Int,
-        newCols: ArrayBuilder[ColumnView]): ColumnView = {
+       cv: ColumnView,
+       precision: Int,
+       toClose: ArrayBuilder[ColumnView]): ColumnView = {
```

Comment:
Curious why `ArrayBuilder` instead of `ArrayBuffer` here?  Typically we've used `ArrayBuffer`.